src/art/skypilot/api.py

```diff
@@ -57,37 +59,49 @@ async def initialize_cluster(
 
         # check if cluster already exists
         cluster_status = await to_thread_typed(
-            lambda: sky.status(cluster_names=[self._cluster_name])
+            lambda: sky.stream_and_get(sky.status(cluster_names=[self._cluster_name]))
         )
         if (
             len(cluster_status) == 0
             or cluster_status[0]["status"] != sky.ClusterStatus.UP
         ):
-            await self._launch_cluster(resources, art_version, env_path)
+            launch_job_id = await self._launch_cluster(resources, art_version, env_path)
+            await asyncio.to_thread(
```

Comment:
I'd prefer to use the `to_thread_typed` helper for all our sync -> async conversions in this file. I realize that in the case of `tail_logs`, `to_thread` already properly returns the type, but I actually prefer the `to_thread_typed` syntax personally.